torch/autograd/profiler.py

```diff
@@ -364,42 +368,83 @@ def __init__(
             use_cuda=False,
             record_shapes=False,
             profile_memory=False,
-            with_stack=False):
+            with_stack=False,
+            use_kineto=False,
+            use_cpu=True):
         self.enabled = enabled
-        self.use_cuda = use_cuda
-        self.function_events = None
         if not self.enabled:
             return
+        self.use_cuda = use_cuda
+        self.function_events = None
         self.entered = False
         self.record_shapes = record_shapes
         self.profile_memory = profile_memory
         self.with_stack = with_stack
+        self.use_kineto = use_kineto
+        self.use_cpu = use_cpu
+        if not self.use_cpu:
+            assert self.use_kineto, \
+                "Device-only events supported only with Kineto (use_kineto=True)"
+
+        self.profiler_kind = None
+        self.kineto_activities = []
+        if self.use_kineto:
```

Comment:
you technically don't need a field on `self` - you can just check `self.kineto_activities` being non-empty below